lib/ansible/plugins/action/sleep.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2017, Dag Wieers <dag@wieers.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from time import sleep
+
+from ansible.module_utils._text import to_text
+from ansible.plugins.action import ActionBase
+
+
+class ActionModule(ActionBase):
+    ''' Sleeps for some time '''
+
+    def run(self, tmp=None, task_vars=dict()):
+        ''' Run the sleep action module '''
+
+        result = super(ActionModule, self).run(tmp, task_vars)
+        result['changed'] = False
+
+        try:
+            minutes = int(self._task.args.get('minutes', 0))
+            seconds = int(self._task.args.get('seconds', 0))
+        except ValueError as e:
+            result['duration'] = 0
+            result['failed'] = True
+            result['msg'] = u"non-integer value given for 'minutes' or 'seconds':\n%s" % to_text(e)
+            return result
+
+        duration = minutes * 60 + seconds
+        if duration < 1:
```

Comment:
I suspect it doesn't matter much in practice, given I suspect any useful sleep would be 5 or 10 sec or more. Just curiosity on my part ;->